src/Http/Http/src/Timeouts/CancellationTokenLinker.cs

```diff
@@ -0,0 +1,19 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Internal;
+
+namespace Microsoft.AspNetCore.Http.Timeouts;
+
+internal class CancellationTokenLinker : ICancellationTokenLinker
+{
+    // Pool for cancellation tokens that cancel the handshake
+    private readonly CancellationTokenSourcePool _ctsPool = new();
+
+    public CancellationTokenSource GetLinkedCancellatinTokeSource(CancellationToken originalToken, TimeSpan timeSpan)
+    {
+        using var cancellationTokenSource = _ctsPool.Rent();
```

Comment:
The `using` here looks wrong. You're returning the cts while it's still being used by the linked cts.